src/UserProfileService.Common/V2/Extensions/HttpContextExtension.cs

```diff
@@ -34,12 +35,22 @@ public static string GetUserId(this HttpContext context, ILogger logger = null)
         if (string.IsNullOrWhiteSpace(userId))
         {
             userId = context.User?.Claims
-                ?.FirstOrDefault(c => c.Type == "sub")
-                ?.Value;
+                            ?.FirstOrDefault(c => c.Type == "sub")
+                            ?.Value;
 
             logger?.LogDebugMessage("Received user id from token: {externalId}.", LogHelpers.Arguments(userId));
         }
 
+        if (string.IsNullOrWhiteSpace(userId))
+        {
+            userId = context.User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.NameIdentifier)
+                            ?.Value;
+
+            logger?.LogDebugMessage(
```

Comment:
This logging message can be used for cases with `claim == "sub"` as well.

Reply:
I userd the same message for the sub claim.